float32 x_pos
float32 y_pos
float32 door_width
bool send_data_flag